### PR TITLE
update max name length in solutions

### PR DESCRIPTION
**Pull request by amiraabouhadid**
### Changes

- changed max name length in deployment model in client
- changed max name length in all deployment dashboard validation
- in solutions that require domain, max name length was changed to 15 chars
- in all other solutions, max name length was changed to 35 chars
### Related Issues

#2364

### Tested Scenarios

- entered a name longer than 35 chars in dashboard deployments

image

### CODE REVIEW

**Review by samaradel — APPROVED** [review]

**Review by 0oM4R — COMMENTED** [review]
image

is there a reason for gitea to be only 15 char ?

**Comment by amiraabouhadid** [comment]
> image
> 
> is there a reason for gitea to be only 15 char ?

good question, for node pilot too @zaelgohary ?

**Comment by zaelgohary** [comment]
> 
> > is there a reason for gitea to be only 15 char ?
> 
> 
> 
> good question, for node pilot too @zaelgohary ?

I'm not sure if node pilot has another restriction rule on its own but you may try to deploy and check if it works.

**Comment by amiraabouhadid** [comment]
@0oM4R  changing gitea and node pilot name max length to 35 and deploying throws this error 

image

image

so won't be changing gitea and node pilot max length from 15 to 35

**Comment by 0oM4R** [comment]
i think we need to handle gatewaynameModal length, i see it got changed [here](https://github.com/threefoldtech/tfgrid-sdk-ts/commit/c5f1bad1c09934f276e88c53980a82e17b3ce3fa#diff-13f46ac7ceefb9f9d1ca1a680d587fccdb952ec2a5c94eea64c2cce762236927R237) 

it is happening with matermost

**Review by 0oM4R — CHANGES_REQUESTED** [review]
i think we need to handle gatewaynameModal length, i see it got changed [here](https://github.com/threefoldtech/tfgrid-sdk-ts/commit/c5f1bad1c09934f276e88c53980a82e17b3ce3fa#diff-13f46ac7ceefb9f9d1ca1a680d587fccdb952ec2a5c94eea64c2cce762236927R237) 

it is happening with matermost 
Screenshot from 2024-10-29 13-32-50

**Comment by amiraabouhadid** [comment]
@0oM4R good catch, got validation error for more than 36 chars indeed
image
it has been set to 35 char max now

**Review by 0oM4R — COMMENTED** [review]
i think we have to update gatewaynameModel to accept the namelength + length of the domain @AhmedHanafy725  can you please provide numbers here ? 

this was already exists but I don't know why it got removed
after manage that please update gitea and node pilot domain names

File: packages/playground/src/weblets/tf_domains.vue
Comment (0oM4R) [inline packages/playground/src/weblets/tf_domains.vue:1]: no need to be 15 here as the name here is the gateway name lets make it 35

image

Comment (amiraabouhadid) [reply packages/playground/src/weblets/tf_domains.vue:1]: done 
image

**Review by 0oM4R — APPROVED** [review]

**Review by samaradel — APPROVED** [review]

**Pull request merged**